sortstructs/priority-sorted-struct.go

```diff
@@ -0,0 +1,146 @@
+package sortstructs
+
+// The default Priority is the last index. The last index is not required in the prioirty list, thus any elements added
+// to the struct will be added to the default priority (last).
+const defaultPrioirty = -1
+
+// PrioritySortedStruct sorts elements by prioirity. Priority lists are unordered.
+// Needs a priority list and a get priority function.
+// If a priority does not exist for an added element, it is added to the lowest priority.
+type PrioritySortedStruct[P comparable, K any] struct {
+	// priorityList is the priority of all the elements, where P is the type used for priority.
+	priorityList [][]P
+	// getPriorityValue returns the value of the priority from the element.
+	getPriorityValue func(el K) P
```

Comment:
Something that might make this a bit clearer, and make the code a bit simpler. You could potentially use a separate interface for k:
```
interface PriorityValue {
    GetPriority()
}
```